enumerable.rb

```diff
@@ -0,0 +1,139 @@
+module MyEnumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    array = is_a?(Range) ? to_a : self
+    i = 0
+    while i < array.length
+      yield(array[i])
+      i += 1
+    end
+    array
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    if is_a?(Range)
+      array = to_array
+      array.length.each do |i|
+        yield(array[i], i)
+      end
+    else
+      i = 0
+      while i < array.length
+        yield(array[i], i)
+        i += 1
+      end
+    end
+    array
+  end
+
+  def my_select
+    my_arr = self.class == Hash ? {} : []
+    if my_arr.class == Hash
+      my_each do |key, value|
+        my_arr[key] = value if !value.nil? && yield(key, value)
+      end
+    elsif is_a?(Range)
+      array = to_a
+      array.my_each do |element|
+        my_arr << element if yield(element)
+      end
+    else
+      my_each do |element|
+        my_arr << element if yield(element)
+      end
+    end
+    my_arr
+  end
+
+  def my_all?(args = nil)
+    condition = true
+    my_each do |i|
+      if condition
+        if args.is_a?(Class)
+          condition = i.is_a?(args) unless args.is_a?(Regexp)
+        elsif args.is_a?(Regexp)
+          condition = !i.match(args).nil? if args.is_a?(Regexp)
+        elsif args
+          condition = i == args
+        elsif block_given?
+          condition = false unless yield(i)
+        else
+          condition = false unless i
+        end
+      end
+    end
+    condition
+  end
+
+  def my_any?(*args)
+    state = false
+    if !args[0].nil?
+      my_each { |i| state = true if args[0] === i }
+    elsif !block_given?
+      my_each { |i| state = true if i }
+    else
+      my_each { |i| state = true if yield(i) }
+    end
+    state
+  end
+
+  def my_none?(args = nil, &block)
+    !my_any?(args, &block)
+  end
+
+  def my_count(args = nil)
+    total = 0
+    if args
+      my_each { |i| total += 1 if i == args }
+    elsif !block_given?
+      total = size
+    elsif !args
+      my_each { |i| total += 1 if yield i }
+    end
+    total
+  end
+
+  def my_map(proc)
+    my_arr = []
+    return my_map { |obj| obj } unless block_given?
```

Comment:
Your method is not working correctly when you pass a block of code like this: `{ proc { |num| num < (5 + 7) / 2 } }`
Please adjust it.

It should return an enumerator when no block is given.